scripts/format.sh

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+
+cd "$(dirname "${BASH_SOURCE:-$0}")" || exit
+ROOT="$(git rev-parse --show-toplevel)"
+
+CLANG_FORMAT_VERSION_REQUIRED="12.0.0"
+CLANG_FORMAT_VERSION=""
+BLACK_VERSION_REQUIRED="21.12b0"
+BLACK_VERSION=""
+MAVEN_VERSION=""
+SHELLCHECK_VERSION_REQUIRED="0.7.1"
```

Comment:
Shall we add some scripts to help users to download shellcheck and related check tools?

Reply:
Only suggestion for now (line: 98) cuz the user's env may have huge difference.

Reply:
Now users can skip part of the formatting when :
1) no changes in his part, so no need to do foramt
2) env problems